File: app/src/main/java/com/coinninja/coinkeeper/util/file/FileProviderUtil.java
package com.coinninja.coinkeeper.util.file;

import android.content.Context;
import android.net.Uri;

import java.io.File;

import androidx.core.content.FileProvider;

public class FileProviderUtil {
    public Uri getUriForFile(Context context, File file) {
        return FileProvider.getUriForFile(context.getApplicationContext(),
                context.getPackageName() + ".provider",
                file
        );
    }
}

File: app/src/main/java/com/coinninja/coinkeeper/util/file/FileProviderUtil.kt
package com.coinninja.coinkeeper.util.file

import android.content.Context
import android.net.Uri

import androidx.core.content.FileProvider
import app.dropbit.annotations.Mockable

import java.io.File

@Mockable
class FileProviderUtil {
    fun getUriForFile(context: Context, file: File): Uri {
        return FileProvider.getUriForFile(context.applicationContext,
                context.packageName + ".provider",
                file
        )
    }
}
